src/lib/db/access-store.ts

```diff
@@ -636,17 +649,54 @@ export class AccessStore implements IAccessStore {
         );
 
         await inTransaction(this.db, async (tx) => {
+            const errors: string[] = [];
             if (groupRows.length > 0) {
                 await tx(T.GROUP_ROLE)
                     .insert(groupRows)
                     .onConflict(['project', 'role_id', 'group_id'])
-                    .merge();
+                    .merge()
+                    .catch((err) => {
+                        if (
+                            err.message.includes(
+                                `violates foreign key constraint "group_role_group_id_fkey"`,
+                            )
+                        ) {
+                            errors.push(
+                                `Your request contains one or more group IDs that do not exist. You sent these group IDs: ${groups.join(
+                                    ', ',
+                                )}.`,
+                            );
+                        }
+                    });
             }
             if (userRows.length > 0) {
                 await tx(T.ROLE_USER)
                     .insert(userRows)
                     .onConflict(['project', 'role_id', 'user_id'])
-                    .merge();
+                    .merge()
+                    .catch((err) => {
```

Comment:
I'm wondering if we handle those types of scenarios consistently across our codebase. Another option would be for the service layer to check if user ID exists before calling a store. On the other handle it would require extra DB call prior to calling the store. With you solution we make 1 DB operation and catch FK violations.

Reply:
Yeah, I thought about that, but I didn't want to start adding extra dependencies for that service for this. If we think it's preferable, then that's a good way to go, though.

Reply:
So in essence: if you think that's better, I'm happy to do that. With the rest of this in place, it should be pretty simple. And we could just move this test to the service layer instead.

Thoughts?

Reply:
I'm fine with the current approach